origin-contracts/contracts/identity/KeyHolderLibrary.sol

```diff
@@ -204,4 +208,13 @@ library KeyHolderLibrary {
         }
         return isThere;
     }
+
+    function _hasNoMatchingPurpose(Key storage key, uint purpose) public view returns (bool hasPurpose) {
```

Comment:
This method could be more efficient if the set of purposes is known ahead of time.